e2e/tests/interchain_accounts/groups_test.go

```diff
@@ -0,0 +1,185 @@
+
+	t.Run("create group with new threshold decision policy", func(t *testing.T) {
+		members := []grouptypes.MemberRequest{
+			{
+				Address: chainAAddress,
+				Weight:  "1",
+			},
+		}
+
+		decisionPolicy := grouptypes.NewThresholdDecisionPolicy("1", time.Duration(time.Minute), time.Duration(0))
+		msgCreateGroupWithPolicy, err := grouptypes.NewMsgCreateGroupWithPolicy(chainAAddress, members, "ics27-controller-group", "ics27-controller-policy", true, decisionPolicy)
+		s.Require().NoError(err)
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, chainAWallet, msgCreateGroupWithPolicy)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(txResp)
+
+	})
+
+	t.Run("submit proposal register interchain account", func(t *testing.T) {
+		groupPolicyAddr = s.QueryGroupPolicyAddress(ctx, chainA)
+		msgRegisterAccount := controllertypes.NewMsgRegisterAccount(ibctesting.FirstConnectionID, groupPolicyAddr, icatypes.NewDefaultMetadataString(ibctesting.FirstConnectionID, ibctesting.FirstConnectionID))
+
+		msgSubmitProposal, err := grouptypes.NewMsgSubmitProposal(groupPolicyAddr, []string{chainAAddress}, []sdk.Msg{msgRegisterAccount}, "", grouptypes.Exec_EXEC_UNSPECIFIED)
+		s.Require().NoError(err)
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, chainAWallet, msgSubmitProposal)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(txResp)
+	})
+
+	t.Run("vote and exec proposal", func(t *testing.T) {
+		msgVote := &grouptypes.MsgVote{
+			ProposalId: 1,
+			Voter:      chainAAddress,
+			Option:     grouptypes.VOTE_OPTION_YES,
+			Exec:       grouptypes.Exec_EXEC_TRY,
+		}
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, chainAWallet, msgVote)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(txResp)
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
+	t.Run("verify interchain account registration success", func(t *testing.T) {
+		interchainAccAddr, err = s.QueryInterchainAccount(ctx, chainA, groupPolicyAddr, ibctesting.FirstConnectionID)
+		s.Require().NoError(err)
+
+		channels, err := relayer.GetChannels(ctx, s.GetRelayerExecReporter(), chainA.Config().ChainID)
+		s.Require().NoError(err)
+		s.Require().Equal(len(channels), 2)
+	})
+
+	t.Run("fund interchain account wallet", func(t *testing.T) {
+		err := chainB.SendFunds(ctx, ibctest.FaucetAccountKeyName, ibc.WalletAmount{
+			Address: interchainAccAddr,
+			Amount:  testvalues.StartingTokenAmount,
+			Denom:   chainB.Config().Denom,
+		})
+		s.Require().NoError(err)
+	})
+
+	t.Run("submit proposal for submit tx", func(t *testing.T) {
+		msgBankSend := &banktypes.MsgSend{
+			FromAddress: interchainAccAddr,
+			ToAddress:   chainBAddress,
+			Amount:      sdk.NewCoins(testvalues.DefaultTransferAmount(chainB.Config().Denom)),
+		}
+
+		cfg := simappparams.MakeTestEncodingConfig()
+		banktypes.RegisterInterfaces(cfg.InterfaceRegistry)
+		cdc := codec.NewProtoCodec(cfg.InterfaceRegistry)
+
```

Comment:
We should add a cdc to our testsuite in a followup pr

Reply:
Definitely agree. https://github.com/cosmos/ibc-go/issues/2512